content/geometry/insidePolygon.h

```diff
@@ -1,34 +1,31 @@
 /**
- * Author: Ulf Lundstrom
- * Date: 2009-03-22
+ * Author: Victor Lecomte, chilli
+ * Date: 2019-04-26
  * License: CC0
- * Source: Basic geometry
- * Description: Returns true if p lies within the polygon described by the points between iterators begin and end. If strict false is returned when p is on the edge of the polygon. Answer is calculated by counting the number of intersections between the polygon and a line going from p to infinity in the positive x-direction. The algorithm uses products in intermediate steps so watch out for overflow. If points within epsilon from an edge should be considered as on the edge replace the line "if (onSegment..." with the comment bellow it (this will cause overflow for int and long long).
+ * Source: https://vlecomte.github.io/cp-geo.pdf
+ * Description: Returns true if p lies within the polygon. If p is on the edge of the polygon, it returns false when strict is true. Answer is calculated by counting the number of intersections between the polygon and a line going from p to infinity in the positive x-direction. The algorithm uses products in intermediate steps so watch out for overflow. Works with integers.
```

Comment:
again, the `Works with integers.` comment is pretty much implicit. The interesting question is whether it works with doubles. It seems to have some special-casing for it, but I'm not quite sure about the handling of parallel lines, I'd need to think about it.

Reply:
Also, can you line-wrap this description?

Reply:
Line wrapped the description.

Also, why is it implicit? It's obvious if you understand that all the operations are just cross/dot products, but I think it's worth noting so you can understand at a glance.

Reply:
All algorithms work with integers if they don't have floating point results, and if it didn't work with integers, it would do `typedef Point<double> P;`.

Reply:
I wasn't sure that was the intended standard, since I saw that `circleTangents` was written like this:
```cpp
template<class P>
pair<P,P> circleTangents(const P &p, const P &c, double r) {
}
```
but had this in the description:

" P is intended to be Point<double>"

Reply:
Oh, we should just update circleTangents in that case. `typedef` is clearer than a description note, I think.